File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/core/InterpolatedSql.kt
package dev.typr.foundations.docs.core

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class InterpolatedSql {
    //start
    val debugListener = object : QueryListener {
        override fun beforeQuery(sql: String, name: String?) {}
        override fun afterQuery(event: QueryEvent) {
            println(event.interpolatedSql())
        }
        override fun failedQuery(event: QueryEvent) {
            System.err.println("Failed: ${event.interpolatedSql()}")
        }
    }
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/InterpolatedSql.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

import dev.typr.foundations.{QueryListener, QueryEvent}

@SuppressWarnings(Array("unused"))
object InterpolatedSql:
  // start
  object debugListener extends QueryListener:
    override def beforeQuery(sql: String, name: java.util.Optional[String]): Unit = ()
    override def afterQuery(event: QueryEvent): Unit =
      println(event.interpolatedSql())
    override def failedQuery(event: QueryEvent): Unit =
      System.err.println(s"Failed: ${event.interpolatedSql()}")
  // stop
